### PR TITLE
H-2346: Add `/me` router that redirects to user's profile page

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

Adds a `/me` route that will redirect an authenticated user to `/@their-shortname`.

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## ❓ How to test this?

1. Visit `/me` while logged in

### CODE REVIEW

**Comment by nonparibus** [comment]
@CiaranMn If I visit while logged in this works great.

If I visit while logged out I get taken to `/signin` rather than what I expected: `/signin?return_to=%2Fme`

But then when I actually test using the latter, while correctly redirecting to `/me` upon login, that page shows the `Profile not found` message that I think is our default 404 (as I have seen it on other made-up routes, as well).

TL;DR: looks like this works upon normal user navigation, but not after a login redirect. And we should make sure logged-out users visiting the route get redirected appropriately.

**Review by nonparibus — CHANGES_REQUESTED** [review]
^

**Comment by CiaranMn** [comment]
> If I visit while logged out I get taken to /signin rather than what I expected: /signin?return_to=%2Fme
@nonparibus I have added this to all redirects to `/signin` (it wasn't present before)

> But then when I actually test using the latter, while correctly redirecting to /me upon login, that page shows the Profile not found message that I think is our default 404 (as I have seen it on other made-up routes, as well).

This happens to me locally, we can have a look sync – it might be something to do with preview deployments / cookies
